src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -1,43 +1,80 @@
 package com.celements.course.service;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 import org.apache.commons.lang.StringUtils;
+import org.apache.velocity.VelocityContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.xwiki.component.annotation.Component;
 import org.xwiki.component.annotation.Requirement;
 import org.xwiki.context.Execution;
+import org.xwiki.model.ModelConfiguration;
 import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.SpaceReference;
 import org.xwiki.model.reference.WikiReference;
 
 import com.celements.common.classes.IClassCollectionRole;
 import com.celements.course.classcollections.CourseClasses;
+import com.celements.course.registration.Person;
+import com.celements.course.registration.RegistrationData;
+import com.celements.mailsender.IMailSenderRole;
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentAlreadyExistsException;
+import com.celements.model.access.exception.DocumentSaveException;
+import com.celements.model.context.ModelContext;
+import com.celements.nextfreedoc.INextFreeDocRole;
+import com.celements.rendering.RenderCommand;
+import com.celements.web.plugin.cmd.ConvertToPlainTextException;
+import com.celements.web.plugin.cmd.PlainTextCommand;
 import com.celements.web.service.IWebUtilsService;
 import com.xpn.xwiki.XWikiContext;
 import com.xpn.xwiki.XWikiException;
 import com.xpn.xwiki.doc.XWikiDocument;
 import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.web.XWikiMessageTool;
+import com.xpn.xwiki.web.XWikiRequest;
 
 @Component
 public class CourseService implements ICourseServiceRole {
 
+  private static Logger LOGGER = LoggerFactory.getLogger(CourseService.class);
+
   @Requirement("CelCourseClasses")
   private IClassCollectionRole courseClasses;
 
   @Requirement
   IWebUtilsService webUtilsService;
 
   @Requirement
+  IModelAccessFacade modelAccess;
+
+  @Requirement
+  ModelContext modelContext;
+
+  @Requirement
+  ModelConfiguration modelConfig;
+
+  @Requirement
+  INextFreeDocRole nextFreeDoc;
+
+  @Requirement
+  IMailSenderRole mailSender;
+
+  @Requirement
   private Execution execution;
 
   private XWikiContext getContext() {
     return (XWikiContext) execution.getContext().getProperty("xwikicontext");
   }
 
   @Override
-  public DocumentReference getCourseTypeForCourse(DocumentReference courseDocRef
-      ) throws XWikiException {
+  public DocumentReference getCourseTypeForCourse(DocumentReference courseDocRef)
+      throws XWikiException {
     DocumentReference typeDocRef = null;
     WikiReference wikiRef = webUtilsService.getWikiRef(courseDocRef);
-    XWikiDocument courseDoc = getContext().getWiki().getDocument(courseDocRef, 
-        getContext());
+    XWikiDocument courseDoc = getContext().getWiki().getDocument(courseDocRef, getContext());
```

Comment:
please refactor the two lines 77,78 to one modelAccess.getXObject(docRef, classRef)